fix: resolve HTTP_2 endpoint behind nginx

gcusnieux opened the pull request:
From [HTTP_2 RFC](https://datatracker.ietf.org/doc/html/rfc7540):

  To ensure that the HTTP/1.1 request line can be reproduced
  accurately, this pseudo-header field MUST be omitted when
  translating from an HTTP/1.1 request that has a request target in
  origin or asterisk form (see [RFC7230], Section 5.3).  Clients
  that generate HTTP/2 requests directly SHOULD use the :authority
  pseudo-header field instead of the Host header field.  An
  intermediary that converts an HTTP/2 request to HTTP/1.1 MUST
  create a Host header field if one is not present in a request by
  copying the value of the :authority pseudo-header field.

We can trust Vertx to add the `:authority` pseudo-header instead of the `Host` header.

gravitee-io/issues#6719
<!-- UI placeholder -->
🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6719-http2-headers-3-5/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)
<!-- UI placeholder end -->

jhaeyaert commented on gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/http/connector/AbstractConnector.java:
```diff
@@ -107,16 +108,21 @@ public ProxyConnection request(ProxyRequest proxyRequest) {
                     url.getHost() : url.getHost() + ':' + port;
-
-            proxyRequest.headers().set(HttpHeaders.HOST, host);
+                proxyRequest.headers().set(HttpHeaders.HOST, host);
```
I'm wondering if we should apply the same logic for both `Host` and `:authority` ?

gcusnieux committed:
fix: resolve HTTP_2 endpoint behind nginx

From HTTP_2 RFC:

To ensure that the HTTP/1.1 request line can be reproduced
accurately, this pseudo-header field MUST be omitted when
translating from an HTTP/1.1 request that has a request target in
origin or asterisk form (see [RFC7230], Section 5.3). Clients
that generate HTTP/2 requests directly SHOULD use the :authority
pseudo-header field instead of the Host header field. An
intermediary that converts an HTTP/2 request to HTTP/1.1 MUST
create a Host header field if one is not present in a request by
copying the value of the :authority pseudo-header field.

We can trust Vertx to add the :authority pseudo-header instead of the Host header.

gravitee-io/issues#6719

gcusnieux committed:
fix: set request into HTTP chunked mode if endpoint use HTTP_2

HTTP_2 doesn't support HTTP 1.1's chunked transfer encoding mechanism, as it provides its own, more efficient, mechanisms for data streaming.

gcusnieux commented:
Replaced by https://github.com/gravitee-io/gravitee-api-management/pull/2064
